start-mycroft.sh

```diff
@@ -131,39 +158,49 @@ function launch-background() {
     fi
 
     # Security warning/reminder for the user
-    if [[ "${1}" == "bus" ]] ; then
+    if [ "${1}" = "bus" ] ; then
         echo "CAUTION: The Mycroft bus is an open websocket with no built-in security"
         echo "         measures.  You are responsible for protecting the local port"
         echo "         8181 with a firewall as appropriate."
     fi
 
-    # Launch process in background, sending logs to standard location
-    python3 -m ${_module} $_params >> /var/log/mycroft/${1}.log 2>&1 &
+    # Launch process in background
+    # Send logs to old standard location if it exists
+    # Otherwise send to XDG Base Directories cache location
+    logdir="/var/log/mycroft/${1}.log"
+    if [ ! -f "$logdir" ]; then
+	    if [ ! -z ${XDG_CACHE_HOME+x} ]; then
```

Comment:
I'm not entirely sure if the cache dir is the right directory. Maybe it should go to `XDG_DATA_HOME` (so `$HOME/.local/share`) instead?

Reply:
Based purely on the [directory descriptions](https://wiki.archlinux.org/index.php/XDG_Base_Directory) on the Arch wiki I'd lean toward `XDG_DATA_HOME`. But a very quick search doesn't seem to give any definitive answers. Might post in Chat and see if anyone has strong opinions

Reply:
The XDG spec designates `XDG_DATA_HOME` for "user-specific data," whereas `XDG_CACHE_HOME` is meant for "user-specific non-essential (cached) data."

By extension, data in `XDG_DATA_HOME` *is* essential.

It's worth mentioning that this question has been asked a million times and still has no clear answer. Some people would like to flush the cache dir from time to time, because it says *cache* on it. You don't want logfiles going *poof* on power loss. On the other hand, some people want to track the data dir in a VCS, because dotfiles. You probably don't want to commit your logs to a VCS.

I vote `XDG_CACHE_HOME`. They're non-essential data, and the likelihood of somebody *actually* flushing `XDG_CACHE_HOME` is much slimmer than somebody tracking `XDG_DATA_HOME` (that'll happen a lot.)

---

At one point, Debian proposed yet another envvar, for a directory to store things like logs and histfiles. That was a long time ago. The XDG mailing list isn't very active. Indeed, the most active thread right now is about who's gonna take over an abandoned spec...

Reply:
Considering /var/log/ in many distributions and / or setups is mounted on a tmpfs and so be gone at a reboot I am with @PureTryOut and @ChanceNCounter classifying it as non-essential data and therefor XDG_CACHE_HOME fits the description better.

Reply:
There are two cache directories defined in the linux file hireachy spec man pages:

SYSTEM Level: /var/cache https://man7.org/linux/man-pages/man7/file-hierarchy.7.html#PERSISTENT_VARIABLE_SYSTEM_DATA

User Level: https://man7.org/linux/man-pages/man7/file-hierarchy.7.html#HOME_DIRECTORY

XDG_CACHE_HOME env var is *generally* defined inside the home directory as per the hierarchy above and is accessible at user level, tho this location might not always be guaranteed since some distros do choose their own standards, but then it's upto that unique distro not following the standard implemention to patch their own paths at packaging level.

Reply:
We're going with XDG_CACHE_HOME, thanks for your opinions guys!